sdks/cpp/lite/include/NamedChoiceConstraint.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+/**
+ * @file NamedChoiceConstraint.h
+ * @brief A constraint that checks if a value is within a named choice
+ * @author isaac.robert@rossvideo.com
+ * @date 2024-07-25
+ * @copyright Copyright (c) 2024 Ross Video
+ */
+
+#include <common/include/IConstraint.h>
+#include <google/protobuf/message_lite.h>
+#include <lite/include/PolyglotText.h>
+
+using catena::common::IConstraint;
+using catena::lite::PolyglotText;
+
+/**
+ * @brief Named choice constraint, ensures a value is within a named choice.
+ */
+template <typename T> 
+class NamedChoiceConstraint : public IConstraint {
+public:
+    /**
+     * @brief Construct a new Named Choice Constraint object
+     * @param init the list of choices
+     */
+    NamedChoiceConstraint(std::vector<std::pair<T, PolyglotText::ListInitializer>> init,
```

Comment:
you may consider adding:

```cpp
using ChoicesInit = std::vector<std::pair<T, PolyglotText::ListInitializer>>; 

```

to that new section of the class def ;-)

or even...

```cpp

using Choices = std::unordered_map<T, PolyglotText>;
using ChoiceInit = std::pair<T, PolyglotText::ListInitializer>;
using ChoicesInit = std::vector<ChoiceInit>;

```

with some helpful doxygen  comments either `/// comment` or `/*< comment */`